packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3235 @@
+    return <Map<String, Object?>>[PlatformMenuItem.serialize(this, delegate, getId)];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(FlagProperty('enabled', value: onSelected != null || onSelectedIntent != null, ifFalse: 'DISABLED'));
+    properties.add(DiagnosticsProperty<FocusNode>('focusNode', focusNode, defaultValue: null));
+    properties.add(DiagnosticsProperty<Widget>('leadingIcon', leadingIcon, defaultValue: null));
+    properties.add(StringProperty('label', label));
+    properties.add(DiagnosticsProperty<Widget>('trailingIcon', trailingIcon, defaultValue: null));
+    properties.add(StringProperty('semanticLabel', semanticLabel, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsets?>('padding', padding, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('foregroundColor', foregroundColor, defaultValue: null));
+    properties
+        .add(DiagnosticsProperty<MaterialStateProperty<Color?>>('overlayColor', overlayColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<TextStyle?>>('textStyle', textStyle, defaultValue: null));
+  }
+}
+
+class _MenuBarItemState extends State<MenuBarItem> {
+  _MenuNode? menu;
+  int menuSerial = 0;
+  late _MenuBarController controller;
+  bool registered = false;
+  Timer? hoverTimer;
+  FocusNode get focusNode => widget.focusNode ?? _focusNode!;
+  FocusNode? _focusNode;
+
+  bool get enabled {
+    return (widget.onSelected != null || widget.onSelectedIntent != null) && controller.enabled;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _focusNode = FocusNode(debugLabel: 'MenuBarItem(${widget.label})');
+  }
+
+  @override
+  void dispose() {
+    hoverTimer?.cancel();
+    hoverTimer = null;
+    controller.unregisterMenu(menu!);
+    _focusNode?.dispose();
+    _focusNode = null;
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
+    _updateMenuRegistration();
+    super.didChangeDependencies();
+  }
+
+  @override
+  void didUpdateWidget(MenuBarItem oldWidget) {
+    if (widget.focusNode != null) {
+      _focusNode?.dispose();
+      _focusNode = null;
+    } else {
+      _focusNode ??= FocusNode(debugLabel: 'MenuBarItem(${widget.label})');
+    }
+    _updateMenuRegistration();
+    super.didUpdateWidget(oldWidget);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final MenuBarThemeData menuBarTheme = MenuBarTheme.of(context);
+    final _TokenDefaultsM3 defaultTheme = _TokenDefaultsM3(context);
+    final Size densityAdjustedSize = const Size(64, 48) + Theme.of(context).visualDensity.baseSizeAdjustment;
+    MaterialStateProperty<EdgeInsets?> resolvedPadding;
```

Comment:
nit: `final` since it gets initialized in all code pathes.

Reply:
Ooh, there should be a lint for that...  Fixed.